src/components/Products/Products.tsx

```diff
@@ -17,37 +22,92 @@ const filterProducts = (products: Product[], filter: FilterType, searchQuery: st
         .filter((product) => !searchQuery || product.title?.toLowerCase().includes(searchQuery.toLowerCase()))
         .filter((product) => selectedCategories.length === 0 || selectedCategories.includes(product.category?.name ?? ''))
         .sort((a, b) => {
-            switch (filter) {
-                case FilterType.PRICE: {
-                    return (b.price ?? 0) - (a.price ?? 0);
-                }
-                case FilterType.NEWEST: {
-                    return new Date(b.creationAt ?? '').getTime() - new Date(a.creationAt ?? '').getTime();
-                }
-                case FilterType.OLDEST: {
-                    return new Date(a.creationAt ?? '').getTime() - new Date(b.creationAt ?? '').getTime();
-                }
-                default: {
-                    return 0;
-                }
+            if (filter === FilterType.PRICE) {
+                return (b.price ?? 0) - (a.price ?? 0);
             }
+            if (filter === FilterType.NEWEST) {
+                return new Date(b.creationAt ?? '').getTime() - new Date(a.creationAt ?? '').getTime();
+            }
+            if (filter === FilterType.OLDEST) {
+                return new Date(a.creationAt ?? '').getTime() - new Date(b.creationAt ?? '').getTime();
+            }
+            return 0;
         });
 
-export const Products: React.FC<{ isDarkTheme: boolean }> = ({ isDarkTheme }) => {
+export const Products = ({ isDarkTheme }: { isDarkTheme: boolean }) => {
+    const [products, setProducts] = useState<Product[]>([]);
     const [currentPage, setCurrentPage] = useState(1);
-    const [filter, setFilter] = useState<FilterType>(FilterType.PRICE);
+    const [filter, setFilter] = useState(FilterType.PRICE);
     const [searchQuery, setSearchQuery] = useState('');
     const [selectedCategories, setSelectedCategories] = useState<string[]>([]);
+    const [isLoading, setIsLoading] = useState(true);
+    const [error, setError] = useState<string | null>(null);
+    const [hasMore, setHasMore] = useState(true);
+    const [isMobile, setIsMobile] = useState(false);
 
-    const filteredItems = useMemo(
-        () => filterProducts(mock, filter, searchQuery, selectedCategories),
-        [filter, searchQuery, selectedCategories],
-    );
+    useEffect(() => {
```

Comment:
move request logic into separate custom hook, or creat e and abstract hook to make the requests

https://dev.to/techcheck/custom-react-hook-usefetch-eid